mmocr/models/textdet/modules/gcn.py

```diff
@@ -0,0 +1,75 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.nn import init
+
+
+class MeanAggregator(nn.Module):
+
+    def forward(self, features, A):
+        x = torch.bmm(A, features)
+        return x
+
+
+class GraphConv(nn.Module):
+
+    def __init__(self, in_dim, out_dim):
+        super(GraphConv, self).__init__()
+        self.in_dim = in_dim
+        self.out_dim = out_dim
+        self.weight = nn.Parameter(torch.FloatTensor(in_dim * 2, out_dim))
+        self.bias = nn.Parameter(torch.FloatTensor(out_dim))
+        init.xavier_uniform_(self.weight)
+        init.constant_(self.bias, 0)
+        self.aggregator = MeanAggregator()
+
+    def forward(self, features, A):
+        b, n, d = features.shape
+        assert d == self.in_dim
+        agg_feats = self.aggregator(features, A)
+        cat_feats = torch.cat([features, agg_feats], dim=2)
+        out = torch.einsum('bnd,df->bnf', cat_feats, self.weight)
+        out = F.relu(out + self.bias)
+        return out
+
+
+class GCN(nn.Module):
+    """Graph convolutional network for clustering. This was from repo
+    https://github.com/Zhongdao/gcn_clustering.
```

Comment:
Mention the orig repo's licence here

Reply:
MIT License. should i mention it in docstring?

Reply:
If there are lots of functions, then you can group them into a file (or many files), and add licence info at the top of the file.